src/twisted/internet/defer.py

```diff
@@ -515,54 +542,151 @@ def addCallbacks(
         #     the caller should treat the result as the new type, consistently.
         return cast(Deferred[_NextDeferredResultT], self)
 
+    @overload
```

Comment:
Was there a reason not to use `@overload` for `addCallbacks` but to use it here?

edit - maybe because there could be as many as ~20 overloads for `addCallbacks`?

Reply:
The fact that you can't specify the argument signatures to the callbacks and errbacks makes it pointless to even attempt overloads there.